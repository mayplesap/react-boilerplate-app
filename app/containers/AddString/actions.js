/*
 * AddString Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  CHANGE_STRING,
  SUBMIT_STRING_SUCCESS,
  SUBMIT_STRING_ERROR,
} from './constants';

/**
 * Changes the input field of the form
 *
 * @return {object} An action object with a type of CHANGE_STRING
 */
export function changeString(input) {
  return {
    type: CHANGE_STRING,
    input,
  };
}

/**
 * Dispatched when the input is submited by the request saga
 *
 * @return {object}      An action object with a type of SUBMIT_STRING_SUCCESS passing the input
 */
export function stringSubmitted(submit) {
  return {
    type: SUBMIT_STRING_SUCCESS,
    submit,
  };
}

/**
 * Dispatched when submitting the input fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of SUBMIT_STRING_ERROR passing the error
 */
export function stringSubmitError(error) {
  return {
    type: SUBMIT_STRING_ERROR,
    error,
  };
}
