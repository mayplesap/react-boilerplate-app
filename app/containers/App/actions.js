/*
 * App Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  LOAD_STRING,
  LOAD_STRING_SUCCESS,
  LOAD_STRING_ERROR,
} from './constants';

/**
 * Load the list, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_STRING
 */
export function loadList() {
  return {
    type: LOAD_STRING,
  };
}

/**
 * Dispatched when the list is loaded by the request saga
 *
 * @return {object}      An action object with a type of LOAD_STRING_SUCCESS passing the repos
 */
export function listLoaded(strings) {
  return {
    type: LOAD_STRING_SUCCESS,
    strings,
  };
}

/**
 * Dispatched when loading the list fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_STRING_ERROR passing the error
 */
export function listLoadingError(error) {
  return {
    type: LOAD_STRING_ERROR,
    error,
  };
}
