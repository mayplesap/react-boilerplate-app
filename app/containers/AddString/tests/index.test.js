/**
 * Test the AddString page
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { AddString, mapDispatchToProps } from '../index';
import { changeString, stringSubmitted } from '../actions';
import configureStore from '../../../configureStore';

describe('<AddString />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddString submit={false} error={false} />
        </IntlProvider>
      </Provider>,
    );

    expect(firstChild).toMatchSnapshot();
  });

  it('should not call onSubmitForm if input is an empty string', () => {
    const submitSpy = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddString onChangeString={() => {}} onSubmitForm={submitSpy} />
        </IntlProvider>
      </Provider>,
    );

    expect(submitSpy).not.toHaveBeenCalled();
  });
});

describe('mapDispatchToProps', () => {
  describe('onChangeString', () => {
    it('should be injected', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      expect(result.onChangeString).toBeDefined();
    });

    it('should dispatch changeString when called', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      const input = 'test';
      result.onChangeString({ target: { value: input } });
      expect(dispatch).toHaveBeenCalledWith(changeString(input));
    });
  });

  describe('onSubmitForm', () => {
    it('should be injected', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      expect(result.onSubmitForm).toBeDefined();
    });

    it('should dispatch stringSubmitted when called', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      const input = 'test';
      result.onSubmitForm({ target: { string: { value: input } } });
      expect(dispatch).toHaveBeenCalledWith(stringSubmitted(input));
    });

    it('should preventDefault if called with event', () => {
      const preventDefault = jest.fn();
      const result = mapDispatchToProps(() => {});
      const input = 'test';
      const evt = { preventDefault, target: { string: { value: input } } };
      result.onSubmitForm(evt);
      expect(preventDefault).toHaveBeenCalledWith();
    });
  });
});
