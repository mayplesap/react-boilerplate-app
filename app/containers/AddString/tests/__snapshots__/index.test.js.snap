// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddString /> should render and match the snapshot 1`] = `
<section
  class="Section-ojdkzv-0 ktJwEy"
>
  <section
    class="Section-ojdkzv-0 CenteredSection-sc-1e2by6n-0 jhRZyk"
  >
    <h1>
      <span>
        This is the AddString container!
      </span>
    </h1>
    <form
      class="Form-sc-148ipml-0 hVyDgC"
    >
      <label
        for="string"
      >
        <span>
          Add new item
        </span>
      </label>
      <input
        class="Input-i9rmlz-0 caCtWU"
        id="string"
        type="text"
        value=""
      />
      <button
        class="Button-sc-1fkoyq4-0 efGMyW"
        type="submit"
      >
        Add
      </button>
    </form>
    <a
      class="A-br8h0y-0 kEqxUH"
      href="/"
    >
      <button
        class="Button-sc-1fkoyq4-0 efGMyW"
        type="button"
      >
        Back
      </button>
    </a>
  </section>
</section>
`;
